examples/mongodsr/src/index.ts

```diff
@@ -1,48 +1,41 @@
 import {
-  DsrRequest,
-  DsrResponse,
-  ConnectionConfig,
+  type DSRRequest,
+  type DSRResponse,
+  type ConnectionConfig,
   Kind,
-  DeleteStatusEvent,
   RequestStatus,
   RequestStatusReason,
-} from '../../src/types'
+} from '@ketch-sdk/ketch-agent'
 import { MongoClient } from 'mongodb'
-async function HandleRequest(req: DsrRequest, conn: ConnectionConfig): Promise<DsrResponse> {
+
+export async function HandleRequest(req: DSRRequest, conn: ConnectionConfig): Promise<DSRResponse> {
+  console.log(req)
   console.log(conn)
   let uri = `mongodb+srv://${conn.username}:${conn.password}@${conn.host}/?retryWrites=true&w=majority`
   console.log(uri)
   const client = new MongoClient(uri)
   try {
     await client.connect()
-    // services.logger.push("Started processing dsr")
-    // console.log()
-    // console.error()
     const database = client.db('sample_restaurants')
     const restaurants = database.collection('restaurants')
-    await restaurants.deleteOne({ owner_email: req.requestBody.subject.email })
+    await restaurants.deleteOne({ owner_email: req.request.subject.email })
     console.log('deleted')
   } catch (error) {
     console.error(error)
   } finally {
     console.log('in finally')
     await client.close()
   }
-  return new Promise<DsrResponse>(resolve => {
-    resolve({
+  return new Promise<DSRResponse>(resolve => {
```

Comment:
This is convoluted. Just return the { without the Promise<...resolve...

Reply:
Fixed.